### PR TITLE
Performance Enhancements

### PR DESCRIPTION
**Pull request by blangley28**
Performance updates. Merging into `rules-mapping` so it doesn't effect the other PR.

Updates include
1. Threading of processing the rules
2. Using FHIR context once in the app level
3. Custom FhirModelResolver which uses the App's FHIR context
4. Utilizing ELM instead of CQL
5. Debug tools to assist in the translation of CQL to ELM
6. Introduction of the CqlUtils class to clean up CQL work

Lots of files were changed but most of them are just using App.getFhirContext() instead of App.FHIR_CTX

### CODE REVIEW

File: src/main/java/org/hl7/davinci/priorauth/ClaimEndpoint.java [hunk 1]
```diff
@@ -287,50 +285,26 @@ private boolean processClaimItems(Bundle bundle, String id, String relatedId) {
...
-        logger.warning("ClaimEndpoint::processClaimItems:unable to write claim item:" + id + "/" + item.getSequence());
+    // Block until all of the threads are done
+    for (ProcessClaimItemTask itemTask : threads.values()) {
+      try {
+        itemTask.getThread().join();
+        if (itemTask.getStatus() != 0)
+          ret = false;
+        logger.fine("ClaimEndpoint::processClaimItems:finsihed processing " + itemTask.getItemName() + ":"
+            + itemTask.getStatus());
+      } catch (InterruptedException e) {
```
Comment (garypgryan) [inline src/main/java/org/hl7/davinci/priorauth/ClaimEndpoint.java:305]: Are there any other kinds of exceptions we should catch?

File: src/main/java/org/hl7/davinci/priorauth/ProcessClaimItemTask.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+        // Add new item to the database
+        if (!App.getDB().write(Table.CLAIM_ITEM, dataMap)) {
+            logger.warning("ClaimEndpoint::processClaimItems:unable to write claim item:" + this.getItemName());
+            ret = false;
+        }
+
+        return ret;
+    }
+}
```
Comment (garypgryan) [inline src/main/java/org/hl7/davinci/priorauth/ProcessClaimItemTask.java:145]: This is very nice and easy to understand!

File: src/main/java/org/hl7/davinci/rules/ModelResolver.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    protected void initialize() {
+        // HAPI has some bugs where it's missing annotations on certain types. This
+        // patches that.
+        this.fhirContext.registerCustomType(AnnotatedUuidType.class);
+
+        // The context loads Resources on demand which can cause resolution to fail in
+        // certain cases
+        // This forces all Resource types to be loaded.
```
Comment (garypgryan) [inline src/main/java/org/hl7/davinci/rules/ModelResolver.java:36]: nice clear comment!

File: src/main/java/org/hl7/davinci/priorauth/ClaimEndpoint.java (hunk 1 above)
Comment (blangley28) [reply src/main/java/org/hl7/davinci/priorauth/ClaimEndpoint.java:305]: This is the only one I found online we should be catching

**Pull request merged**